data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/prepper/StringPrepper.java

```diff
@@ -57,16 +69,38 @@ public StringPrepper(final Configuration configuration) {
     }
 
     @Override
-    public Collection<Record<String>> execute(final Collection<Record<String>> records) {
-        final Collection<Record<String>> modifiedRecords = new ArrayList<>(records.size());
-        for (Record<String> record : records) {
-            final String recordData = record.getData();
-            final String newData = upperCase? recordData.toUpperCase() : recordData.toLowerCase();
-            modifiedRecords.add(new Record<>(newData));
+    public Collection<Record<Event>> execute(final Collection<Record<Event>> records) {
+        final Collection<Record<Event>> modifiedRecords = new ArrayList<>(records.size());
+        for (Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+            final String eventJson = recordEvent.toJsonString();
+            try {
+                final Map<String, Object> newData = processEventJson(eventJson);
+                final Event newRecordEvent = JacksonEvent.builder()
+                        .withEventMetadata(recordEvent.getMetadata())
+                        .withData(newData)
+                        .build();
+                modifiedRecords.add(new Record<>(newRecordEvent));
+            } catch (JsonProcessingException e) {
+                LOG.error("Unable to process Event data: {}", eventJson, e);
+            }
         }
         return modifiedRecords;
     }
 
+    private Map<String, Object> processEventJson(final String data) throws JsonProcessingException {
```

Comment:
We probably need some new features in `Event` if this is the necessary approach! But, this is fine for this processor.

Reply:
Agree. Did not find an API to iterate through the keys in Event data, neither the Event::toMap() method.